src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -305,36 +305,28 @@ namespace levin
     struct fluff_flush
     {
       std::shared_ptr<detail::zone> zone_;
-      std::chrono::steady_clock::time_point flush_time_;
 
       static void queue(std::shared_ptr<detail::zone> zone, const std::chrono::steady_clock::time_point flush_time)
       {
         assert(zone != nullptr);
         assert(zone->strand.running_in_this_thread());
 
         detail::zone& this_zone = *zone;
-        this_zone.flush_time = flush_time;
+        ++this_zone.flush_callbacks;
         this_zone.flush_txs.expires_at(flush_time);
-        this_zone.flush_txs.async_wait(this_zone.strand.wrap(fluff_flush{std::move(zone), flush_time}));
+        this_zone.flush_txs.async_wait(this_zone.strand.wrap(fluff_flush{std::move(zone)}));
       }
 
       void operator()(const boost::system::error_code error)
       {
-        if (!zone_ || !zone_->p2p)
+        if (!zone_ || --zone_->flush_callbacks || !zone_->p2p)
```

Comment:
That can underflow on error (ie, flush_callbacks is 0 upon entry), which will presumably have bad effects, so an error message would be good.

Reply:
Underflow can only occur if ASIO invokes a queued callback multiple times. This should never happen, but all of the callbacks already have defensive code. So I added a zero check before the decrement.